jobclient/python/__init__.py

```diff
@@ -0,0 +1,206 @@
+        :param env: Environment variables to set within the job's context,
+            defaults to None.
+        :type env: Dict[str, str], optional
+        :param labels: Labels to assign to the job, defaults to None.
+        :type labels: Dict[str, str], optional
+        :param max_runtime: The maximum number seconds this job should be
+            allowed to run, defaults to None.
+        :type max_runtime: int, optional
+        :param name: A name to assign to the job, defaults to None.
+        :type name: str, optional
+        :param priority: A priority to assign to the job, defaults to None.
+        :type priority: int, optional
+        :param application: Application information to assign to the job,
+            defaults to None.
+        :type application: Application, optional
+        Output
+        ------
+        :return: The UUID of the newly-created job.
+        :rtype: UUID
+        """
+        uuid = str(uuid if uuid is not None else util.make_temporal_uuid())
+        payload = {
+            'command': command,
+            'cpus': cpus,
+            'mem': mem,
+            'uuid': uuid,
+            'max-retries': max_retries
+        }
+        if env is not None:
+            payload['env'] = env
+        if labels is not None:
+            payload['labels'] = labels
+        if max_runtime is not None:
+            payload['max-runtime'] = max_runtime
+        if name is not None:
+            payload['name'] = name
+        if priority is not None:
+            payload['priority'] = priority
+        if application is not None:
+            payload['application'] = application.to_dict()
+        payload = {'jobs': [payload]}
+        url = urlunparse(('http', self.__netloc, self.__job_endpoint, '', '', ''))  # noqa E501
+        _LOG.debug(f"Sending POST to {url}")
+        _LOG.debug("Payload:")
+        _LOG.debug(json.dumps(payload, indent=4))
+        resp = requests.post(url, json=payload)
+        if not resp.ok:
+            _LOG.error(f"Could not submit job: {resp.status_code} {resp.text}")
+            resp.raise_for_status()
+
+        return UUID(uuid)
+
+    def query(self, uuid: UUID) -> Job:
+        """Query Cook for a job's status.
+
+        :param uuid: The UUID to query.
+        :type uuid: UUID
+        :return: A Job object containing the job's information.
+        :rtype: Job
+        """
+        if self.__job_endpoint == '/jobs':
+            param_name = 'uuid'
+        else:
+            param_name = 'job'
+        query = urlencode([(param_name, str(uuid))])
+        url = urlunparse(('http', self.__netloc, self.__job_endpoint, '',
+                          query, ''))
+        _LOG.debug(f'Sending GET to {url}')
+        resp = requests.get(url)
+        if not resp.ok:
+            _LOG.error(f"Could not query job: {resp.status_code} {resp.text}")
+            resp.raise_for_status()
+        return Job.from_dict(resp.json()[0])
+
+    def kill(self, uuid: UUID):
```

Comment:
Same here, I think allowing string would be nice.

Reply:
:+1: